lib/charms/traefik_k8s/v0/ingress_unit/requires.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Requires side of ingress-unit relation."""
+
+import logging
+from functools import cached_property
+from pathlib import Path
+
+import sborl
+from ops.charm import CharmBase
+
+logger = logging.getLogger(__name__)
+
+
+class IngressUnitRequirer(sborl.EndpointWrapper):
```

Comment:
It has [the events from the base class](https://github.com/canonical/sborl/blob/main/sborl/events.py#L24-L27). I added the `request` event to the provider because it seemed more semantically intuitive than `ready`; should I do something similar for the requirer?  Something like `urls_ready`?